automerge/src/op_observer.rs

```diff
@@ -1,50 +1,105 @@
 use crate::exid::ExId;
+use crate::Parents;
 use crate::Prop;
 use crate::Value;
 
 /// An observer of operations applied to the document.
-pub trait OpObserver {
+pub trait OpObserver: Default + Clone {
```

Comment:
I don't think these constraints should be here, we should add the constriants where we need them, i.e. in `impl` blocks.